app/bundles/LeadBundle/Entity/LeadDeviceRepository.php

```diff
@@ -98,4 +98,31 @@ public function getDevice($lead, $deviceName = null, $deviceBrand = null, $devic
 
         return (!empty($device)) ? $device[0] : [];
     }
+
+    /**
+     * @param string $fingerprint
+     *
+     * @return LeadDevice
+     */
+    public function getDeviceByFingerprint($fingerprint)
+    {
+        if (!$fingerprint) {
+            return null;
+        }
+
+        $sq = $this->_em->getConnection()->createQueryBuilder();
+        $sq->select('es.id as id, es.lead_id as lead_id')
+            ->from(MAUTIC_TABLE_PREFIX.'lead_devices', 'es');
+
+        if ($fingerprint !== null) {
```

Comment:
This if statement will always be true because the if statement at line 109 won't let $fingerprint to be null.